File: backend/app/schemas/ai_assistant.py
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import BaseModel, Field


class AiAssistantHistoryTurn(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(..., max_length=4000)


class AiAssistantMessageRequest(BaseModel):
    session_id: str = Field(..., min_length=8, max_length=100)
    history: List[AiAssistantHistoryTurn] = Field(default_factory=list, max_length=40)
    message: str = Field(default="", max_length=2000)
    probe: bool = False


class AiAssistantMessageResponse(BaseModel):
    reply: str
    disabled: bool
    disabled_reason: Optional[str] = None
    turns_remaining: Optional[int] = None


class AiAssistantStatusResponse(BaseModel):
    configured: bool
    enabled: bool
    disabled_reason: Optional[str] = None
    disabled_at: Optional[datetime] = None
    disabled_detail: Optional[str] = None
    consecutive_error_count: int
    daily_message_count: int
    daily_message_budget: int


class AiAssistantReenableResponse(BaseModel):
    success: bool
    reason: Optional[str] = None
    detail: Optional[str] = None

File: frontend/src/api/aiAssistant.ts
import { api } from "./index";

export interface AiAssistantHistoryTurn {
  role: "user" | "assistant";
  content: string;
}

export interface AiAssistantMessageRequest {
  session_id: string;
  history: AiAssistantHistoryTurn[];
  message: string;
  probe: boolean;
}

export interface AiAssistantMessageResponse {
  reply: string;
  disabled: boolean;
  disabled_reason: string | null;
  turns_remaining: number | null;
}

export interface AiAssistantStatusResponse {
  configured: boolean;
  enabled: boolean;
  disabled_reason: string | null;
  disabled_at: string | null;
  disabled_detail: string | null;
  consecutive_error_count: number;
  daily_message_count: number;
  daily_message_budget: number;
}

export interface AiAssistantReenableResponse {
  success: boolean;
  reason: string | null;
  detail: string | null;
}

export const aiAssistantApi = {
  sendMessage: (data: AiAssistantMessageRequest) =>
    api.post<AiAssistantMessageResponse>("/public/ai-assistant/message", data),
  status: () => api.get<AiAssistantStatusResponse>("/integrations/ai-assistant/status"),
  reenable: () => api.post<AiAssistantReenableResponse>("/integrations/ai-assistant/reenable"),
};
